DQMOffline/Trigger/plugins/LepHTMonitor.cc

```diff
@@ -0,0 +1,488 @@
+
+  //num and den hists to be divided in harvesting step to make turn on curves
+  h_pfHTTurnOn_num_ = ibooker.book1D("pfHTTurnOn_num", "Numerator;Offline H_{T} [GeV];",  f_htbins.size()-1, f_htbins.data());
+  h_pfHTTurnOn_den_ = ibooker.book1D("pfHTTurnOn_den","Denominator;Offline H_{T} [GeV];",  f_htbins.size()-1, f_htbins.data());
+  
+  h_lepPtTurnOn_num_  = ibooker.book1D("lepPtTurnOn_num", ("Numerator;Offline "+lepton+" p_{T} [GeV];").c_str(),   f_ptbins.size()-1, f_ptbins.data());
+  h_lepPtTurnOn_den_  = ibooker.book1D("lepPtTurnOn_den", ("Denominator;Offline "+lepton+" p_{T} [GeV];").c_str(), f_ptbins.size()-1, f_ptbins.data());
+  h_lepEtaTurnOn_num_ = ibooker.book1D("lepEtaTurnOn_num", "Numerator;Offline lepton #eta;", nbins_eta_,etabins_min_,etabins_max_);
+  h_lepEtaTurnOn_den_ = ibooker.book1D("lepEtaTurnOn_den", "Denominator;Offline lepton #eta;", nbins_eta_,etabins_min_,etabins_max_);
+  h_lepPhiTurnOn_num_ = ibooker.book1D("lepPhiTurnOn_num", "Numerator;Offline lepton #phi;", nbins_phi_,phibins_min_,phibins_max_);
+  h_lepPhiTurnOn_den_ = ibooker.book1D("lepPhiTurnOn_den", "Denominator;Offline lepton #phi;", nbins_phi_,phibins_min_,phibins_max_);
+
+  h_lepEtaPhiTurnOn_num_ = ibooker.book2D("lepEtaPhiTurnOn_num", "Numerator;Offline lepton #eta;Offline lepton #phi;", nbins_eta_/2,etabins_min_,etabins_max_,nbins_phi_/2,phibins_min_,phibins_max_);
+  h_lepEtaPhiTurnOn_den_ = ibooker.book2D("lepEtaPhiTurnOn_den", "Denominator;Offline lepton #eta;Offline lepton #phi;", nbins_eta_/2,etabins_min_,etabins_max_,nbins_phi_/2,phibins_min_,phibins_max_);
+
+  h_NPVTurnOn_num_  = ibooker.book1D("NPVTurnOn_num", "Numerator;N_{PV};", nbins_npv_,npvbins_min_,npvbins_max_);
+  h_NPVTurnOn_den_  = ibooker.book1D("NPVTurnOn_den", "Denominator;N_{PV};", nbins_npv_,npvbins_min_,npvbins_max_);
+
+  ibooker.cd();
+}
+
+void LepHTMonitor::beginLuminosityBlock(const edm::LuminosityBlock &lumiSeg,
+                                                 const edm::EventSetup &context){
+  edm::LogInfo("LepHTMonitor") << "LepHTMonitor::beginLuminosityBlock\n";
+}
+
+void LepHTMonitor::analyze(const edm::Event &e, const edm::EventSetup &eSetup){
+  edm::LogInfo("LepHTMonitor") << "LepHTMonitor::analyze\n";
+
+  //Find whether main and auxilliary triggers fired
+  bool hasFired = false;
+  bool hasFiredAuxiliary = false;
+  bool hasFiredLeptonAuxiliary = false;
+  if (den_lep_genTriggerEventFlag_->on() && den_lep_genTriggerEventFlag_->accept( e, eSetup) ) hasFiredLeptonAuxiliary=true;
+  if (den_HT_genTriggerEventFlag_->on() && den_HT_genTriggerEventFlag_->accept( e, eSetup) ) hasFiredAuxiliary=true;
+  if (num_genTriggerEventFlag_->on() && num_genTriggerEventFlag_->accept( e, eSetup) ) hasFired=true;
+
+  if(!(hasFiredAuxiliary || hasFiredLeptonAuxiliary)) return;
+  int npv=0;
+  //Vertex
+  edm::Handle<reco::VertexCollection> VertexCollection;
+  if(theVertexCollectionTag_.label() != ""){
```

Comment:
to avoid adding extra variables, I propose to use `if (not theVertexCollectionTag_.label().empty())` which is just a pointer comparison, and should be as fast a checking a bool.